### ISSUE TITLE
[typescript-resolvers] Field wrapper doesn't wrap array types

### ISSUE THREAD

**Issue by nebbles** [dotansimha/graphql-code-generator#5227]
**Describe the bug**
The change made for #3882 is counter-intuitive to me and seems like a bug. By my understanding, a Field definition includes whether its an array or not.

Based on this change, when adding to `config`:
```yml
wrapFieldDefinitions: true
fieldWrapperValue: T | Promise<T> | (() => T) | (() => Promise<T>)
```

Then I'd *expect* this wrapper to be applied to the Type of the field

```ts
// a type like this
images: Array<GqlImageData>;

// turns into
images: FieldWrapper<Array<GqlImageData>>;
```

However as mentioned, the above change means that Arrays cannot be given a custom wrapper. The result of using the config returns

```ts
images: Array<FieldWrapper<GqlImageData>>;
```

Is this a bug or a feature request? As alluded to in the example config, it would simplify code a lot if a custom field wrapper enabled resolver functions for fields (I've read #1704 - however moving to partials removes type strictness which is useful to developers in the team).

Thanks

**Comment by ibratoev** [comment]
I hit the same issue as I use the same `fieldWrapperValue`. Added a few comments in #3882 about it. In my case I get the error `Type '() => Promise<User[]>' is missing the following properties from type 'FieldWrapper<User>[]': pop, push, concat, join, and 27 more`.

**Comment by hampusborgos** [comment]
I have attempted to solve this in a new PR: https://github.com/dotansimha/graphql-code-generator/pull/5753 that adds a new configuration option target to arrays (and `Maybe<>`) wrappers.

**Comment by dotansimha** [comment]
Fix is available in `@graphql-codegen/typescript@1.22.0`. Thanks @hampusborgos !
